crates/angr/src/segmentlist.rs

```diff
@@ -176,17 +182,20 @@ impl SegmentList {
         if size == 0 {
             return;
         }
-        self.0.insert(address..(address + size), sort);
+        self.map.insert(address..(address + size), sort);
+        self.bytes_occupied += size;
```

Comment:
This is only correct if the entire range was unoccupied, same with `update` below

Reply:
A cache in a RwLock would allow avoiding recalculation and eager calculation

Reply:
Or an AtomicU64 to avoid the lock